src/Components/Server/src/Circuits/ICircuitInboundEventHandler.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits;
+
+/// <summary>
+/// A handler to process inbound circuit events.
+/// </summary>
+public interface ICircuitInboundEventHandler
+{
+    /// <summary>
+    /// Invoked when inbound event on the circuit causes an asynchronous task to be dispatched on the server.
+    /// </summary>
+    /// <param name="context">The <see cref="CircuitInboundEventContext"/>.</param>
+    /// <param name="next">The next handler to invoke.</param>
+    /// <returns>A <see cref="Task"/> that completes when the event has finished.</returns>
+    Task HandleInboundEventAsync(CircuitInboundEventContext context, CircuitInboundEventDelegate next);
```

Comment:
This isn't an `in` parameter because that prevents implementors from making the method `async`. Should we defensively make `CircuitInboundEventContext` a `class` to avoid the cost of copying in case we increase the size of this struct in the future?

Reply:
We can make this a class